CODING_GUIDELINES.md

```diff
@@ -139,6 +139,28 @@ Temporarily it may be allowed to split unit tests for a module into multiple `te
 
 Temporarily, tests for testing utilities can be placed next to the module containing them, with the name `test_util_<module>.py`. This should be taken as a hint that the tested utils should be moved into the library.
 
+#### Integration Tests Utils
+
+Integrations tests come with their own utilities, found in `cases.py`, for better test automation and simplification:
+
+    - Field type annotations, e.g. `IJKField = Field[[IDim, JDim, KDim], np.int64]`.
+    - Test fixtures: `cartesian_case` for structured, e.g. `IDim` x `JDim`; `unstructured_case`, e.g. `EdgeDim`.
+    - Parameter allocations for objects, derived directly from the decorator's function definition:
+        ```
+           input_param = cases.allocate(case_fixture, decorator_name, "input_label")()
+        ```
+    - Data initialization/modification classes, such as `ConstInitializer()`, to be used as `allocate(...).strategy()` parameters
+    - Decorator verification functions:
+        - `cases.verify()`: used when input and output field generation cannot be automated. For example the case when the `extend()` method needs to be used for an offset or sparse fields are part of the input dataset.
+        - `cases.verify_with_default_data()`: used when input and output fields generation can be automated.
+    - Backends are set automatically with default switched off. However, if explication is needed, they can be explicated as cases attributes.
+
+In case new test utils need to be written, it is important to first check whether any utilities can be used or modified.
+If this is not the case but the new utility supports similar functionalities as others, it should be placed in the same file. Naming convention should mirror other cases and reflect the purpose.
+In case that the new utility is a completely stand-alone component, a new file should be created in the bottom-most directory enclosing all usages of the utility.
```

Comment:
```suggestion
In case new features are needed, check if an existing utility can be extended before implementing a whole new one. The name of new utilities should reflect their purpose and follow the naming conventions of existing tools. If the new utility is a completely stand-alone component, create a new file in the bottom-most directory enclosing all usages of the utility; otherwise, place it in the appropriate file containing utilities with similar functionalities.
```